Source/Core/Core/IOS/Device.h

```diff
@@ -214,4 +221,20 @@ class Device
 private:
   std::optional<IPCReply> Unsupported(const Request& request);
 };
+
+// Helper class for Devices that we know are only ever instantiated under an EmulationKernel.
+// Deriving a Device from this allows it to access EmulationKernel-only features without runtime
+// overhead, since it will know that the m_ios instance is an EmulationKernel.
+class EmulationDevice : public Device
```

Comment:
`GetEmulationKernel().GetSystem()` is such a common op I wonder if it's worth adding a  `GetSystem()` helper function to EmulationDevice

Reply:
Would probably help readability, yeah.

Reply:
I've added this.